internal/driver/glfw/window.go

```diff
@@ -430,17 +430,29 @@ func (w *window) Close() {
 		return
 	}
 
-	w.viewLock.Lock()
-	w.closing = true
-	w.viewLock.Unlock()
-	w.viewport.SetShouldClose(true)
+	// set w.closing flag inside draw thread to ensure we can free textures
+	runOnDraw(w, func() {
+		w.viewLock.Lock()
+		w.closing = true
+		w.viewLock.Unlock()
+		w.viewport.SetShouldClose(true)
+		cache.RangeTexturesFor(w.canvas, func(obj fyne.CanvasObject) {
+			w.canvas.Painter().Free(obj)
+		})
+	})
 
+	// destroy current renderers
```

Comment:
I'm not sure this (and other comments) are required.
Comments generally show that the code could be clearer - so we refactor.
The in-method comments should be kept for those lines that will need an explanation.

Reply:
Got it, sorry for them, I forgot to remove them. I will make the changes :)

Reply:
done in ad7f974